### ISSUE TITLE
mergeLonghand lowercases values with multiple var references

### ISSUE THREAD

**Issue by lojjic** [cssnano/cssnano#847]
This could be considered a reopening of #648, or a bug with the implemented fix in #649. That did fix simple cases but it still fails when values contain more than a single `var(--X)` reference. An example:

```
border: 1px solid hsla(var(--HUE), var(--SATURATION), var(--LUMINANCE), 0.5)
```

incorrectly transforms to:

```
border: 1px solid hsla(var(--hue), var(--saturation), var(--LUMINANCE), 0.5)
```

The culprit appears to be the greedy leading `.*` in the regular expression at https://github.com/cssnano/cssnano/blob/3580e7be5a4d32ed2b122d13b87380d2b150c705/packages/postcss-merge-longhand/src/lib/parseWsc.js#L6

This toLower logic could probably be fixed to handle strings with multiple vars. However, I'd question whether lowercasing should be happening in parseWsc.js at all. I feel like the mergeLonghand plugin should be limiting itself to only merging, and leave lowercasing up to other specialized plugins that can handle that in a more robust way.

**Comment by anikethsaha** [comment]
@evilebottnawi we can drop the lowercasing in the merge plugin. It can be unsafe

**Comment by evilebottnawi** [comment]
@anikethsaha we should improve this, it is invalid remove lowercase

**Comment by anikethsaha** [comment]
> @anikethsaha we should improve this, it is invalid remove lowercase

Why ? doing lowercase won't change the performance right?
Also, it can be tricky here to handle all the edge cases. 
For this, I can put a loop to check all the `var`, but I am not getting what changes it will bring to the performance? 
Or it can be implemented as an opt-in method. Like passing an option to explicitly doing this.

What do you think?

**Comment by evilebottnawi** [comment]
@anikethsaha we can't ignore lowercase because CSS is case insensitivity in that case, we need improve handling or maybe ignore lines with `var`/`env`/`contant`

**Comment by anikethsaha** [comment]
Does they effect the performance?

**Comment by evilebottnawi** [comment]
@anikethsaha what do you mean? Using lowercase + uppercase notations increase gzip size so it is preferable write all in lowercase notation

**Comment by anikethsaha** [comment]
> Using lowercase + uppercase notations increase gzip size so it is preferable write all in lowercase notation

Ohh yeah this is what I wanted to know. 
I will try to fix it as soon as possible 👍

**Comment by anikethsaha** [comment]
handled in https://github.com/cssnano/cssnano/pull/867

In the meantime use `postcss-lowercase-text` plugin to solve this issue

Once https://github.com/cssnano/cssnano/pull/867 is merged, I will submit another PR to remove the lowercasing from this plugin as it will be handled in separate plugin

**Comment by evilebottnawi** [comment]
This issue should be solved inside plugin without `postcss-lowercase-text`, because some developers can disable some plugins

**Comment by evilebottnawi** [comment]
So we should solve that problem inside plugin

**Comment by anikethsaha** [comment]
if lowercasing is done using a separate plugin, why to do lower casing in other plugins as well. If will be useful to enable/disable at project level.

**Comment by evilebottnawi** [comment]
Because each plugin independent and should solve its problems without the help of other plugins

**Comment by anikethsaha** [comment]
> Because each plugin independent and should solve its problems without the help of other plugins

yeah, so merge long hand plugin should not do any lowercasing right ?

**Comment by evilebottnawi** [comment]
Ideally - yes. But it is often increase memory usage and increase complex of code so a plugin potentially can change some values, but only when it is safe

**Comment by anikethsaha** [comment]
okay 👍 
I think a fix would be to ignore lowercasing when `var` or `calc` comes up.

**Comment by evilebottnawi** [comment]
Yes, we should do not lowercase values inside `var` or `calc`

**Comment by ludofischer** [comment]
Still broken on 5.0.0-rc.2
